src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs

```diff
@@ -893,5 +893,27 @@ public void PasswordBox_With_Description()
 			using var screenshot = TakeScreenshot("PasswordBox Description", new ScreenshotOptions() { IgnoreInSnapshotCompare = true });
 			ImageAssert.HasColorAt(screenshot, passwordBoxRect.X + passwordBoxRect.Width / 2, passwordBoxRect.Y + passwordBoxRect.Height - 50, Color.Red);
 		}
+
+		[Test]
+		[AutoRetry]
+		public void TextBox_Visibility()
+		{
+			Run("UITests.Windows_UI_Xaml_Controls.TextBox.TextBox_Visibility", skipInitialScreenshot: true);
+			var textBoxRect = ToPhysicalRect(_app.WaitForElement("MyTextBox")[0].Rect);
+			var buttonRect = ToPhysicalRect(_app.WaitForElement("ShowHideButton")[0].Rect);
+
+			_app.FastTap("ShowHideButton");
+
+			var textBoxRect2 = ToPhysicalRect(_app.WaitForElement("MyTextBox")[0].Rect);
+			var buttonRect2 = ToPhysicalRect(_app.WaitForElement("ShowHideButton")[0].Rect);
+
+			// Assert that after clicking, MyTextBox goes out of screen (hidden).
+			Assert.AreEqual(-100000 + textBoxRect.X, textBoxRect2.X);
+			Assert.AreEqual(-100000 + textBoxRect.Y, textBoxRect2.Y);
```

Comment:
It's likely that this is not going to work on non-Wasm platforms, it's probably best to validate that the screenshot is actually different.

Reply:
I already assert the the button moved up, so if that passed, we already know the screenshot is different.

I'll try to adjust the problematic assertion for Wasm and other platforms so that it passes. If it doesn't, I think it can just be deleted and the second assertion should keep us safe.